lntest/btcd.go

```diff
@@ -0,0 +1,70 @@
+package lntest
+
+import (
+	"encoding/hex"
+	"fmt"
+
+	"github.com/btcsuite/btcd/chaincfg"
+	"github.com/btcsuite/btcd/integration/rpctest"
+	"github.com/btcsuite/btcd/rpcclient"
+)
+
+// BtcdBackendConfig is an implementation of the BackendConfig interface
+// backed by a btcd node.
+type BtcdBackendConfig struct {
+	// RPCConfig houses the connection config to the backing btcd instance.
+	RPCConfig rpcclient.ConnConfig
+
+	// Harness is the backing btcd instance.
+	Harness *rpctest.Harness
+}
+
+// GenArgs returns the arguments needed to be passed to LND at startup for
+// using this node as a chain backend.
+func (b BtcdBackendConfig) GenArgs() []string {
+	var args []string
+	encodedCert := hex.EncodeToString(b.RPCConfig.Certificates)
+	args = append(args, "--bitcoin.node=btcd")
+	args = append(args, fmt.Sprintf("--btcd.rpchost=%v", b.RPCConfig.Host))
+	args = append(args, fmt.Sprintf("--btcd.rpcuser=%v", b.RPCConfig.User))
+	args = append(args, fmt.Sprintf("--btcd.rpcpass=%v", b.RPCConfig.Pass))
+	args = append(args, fmt.Sprintf("--btcd.rawrpccert=%v", encodedCert))
+
+	return args
+}
+
+// P2PAddr returns the address of this node to be used when connection over the
+// Bitcoin P2P network.
+func (b BtcdBackendConfig) P2PAddr() string {
+	return b.Harness.P2PAddress()
+}
+
+// NewBtcdBackend starts a new rpctest.Harness and returns a BtcdBackendConfig
+// for that node.
+func NewBtcdBackend() (*BtcdBackendConfig, func(), error) {
```

Comment:
Nice! We use something similar in the `chainntnfs` package now. Perhaps we can unify these or have them call the `lntest` package directly? https://github.com/lightningnetwork/lnd/blob/master/chainntnfs/test_utils.go

Reply:
Looks like those are using the `notifier` interface directly, while here we are really configuring `lnd` to start with one of the selected backends. Don't see immediately how we could unify these easily.